test/controllers/PrintControllerSpec.scala

```diff
@@ -94,6 +101,15 @@ class PrintControllerSpec extends UnitSpec with WithFakeApplication with Mockito
         jsoupDoc.body.getElementsByTag("h1").text shouldEqual Messages("Testy Mctestface")
       }
     }
-  }
 
+    "InValid data is provided" should {
+      lazy val result = await(TestPrintControllerInValidDetails.printView(fakeRequest))
+      lazy val jsoupDoc = Jsoup.parse(bodyOf(result))
+      "return " in {
+        status(result) shouldBe 303
```

Comment:
This looks good but could you add a test for the redirect location?

(It works the same way as status: "redirectLocation(result) shouldBe <route>" which requires the import: import play.api.test.Helpers._)

Reply:
Done! can you review again please?